paf-mvp-core-js/src/validation/json-validator.ts

```diff
@@ -0,0 +1,203 @@
+        };
+      } else {
+        const errors = ajvValidate.errors.map(this.buildError);
+        return {
+          isValid: false,
+          errors: errors,
+        };
+      }
+    } catch (error) {
+      return {
+        isValid: false,
+        errors: [error],
+      };
+    }
+  }
+
+  async start() {
+    const schemas = await this.schemaRepository.all();
+    this.ajv = new AjvClass({ strict: false, schemas });
+  }
+
+  private buildError(err: ErrorObject): Error {
+    const pathStr = err.instancePath === '' ? 'in root object' : `at path: ${err.instancePath}`;
+    const msg = `${err.message} (${pathStr}).`;
+    return new JsonSchemaError(msg);
+  }
+}
+
+export class JsonSchemaRepository implements IJsonSchemaRepository {
+  private logger: Log;
+  private schemasDirectory: string;
+
+  public static default(): JsonSchemaRepository {
+    const dir = path.join(__dirname, '..', '..', '..', 'addressability-framework', 'mvp-spec', 'json-schemas');
+    return this.build(dir);
+  }
+
+  public static build(directory: string): JsonSchemaRepository {
+    if (!fs.existsSync(directory)) {
+      throw new Error(`File doesn't exist. Cannot load: ${directory}`);
+    }
+    return new JsonSchemaRepository(directory);
+  }
+
+  async all(): Promise<Json[]> {
+    const fileNames: string[] = await this.listFiles(this.schemasDirectory);
+    const contentPromises: Promise<string>[] = fileNames.map((f) => this.loadFile(f));
+    const contents: string[] = await Promise.all(contentPromises);
+    const jsons: Json[] = this.createJsons(contents);
+    this.cleanUp(jsons);
+    return jsons;
+  }
+
+  private constructor(directory: string) {
+    this.logger = new Log('JsonSchemaRepository', '#bbb');
+    this.schemasDirectory = directory;
+  }
+
+  private async listFiles(dir: string): Promise<string[]> {
+    const files = await fs.promises.readdir(dir);
+    const filtered = files.filter((f: string) => f.endsWith('.json'));
+    return filtered;
+  }
+
+  private async loadFile(file: string): Promise<string> {
+    const fullPath = path.join(this.schemasDirectory, file);
+    const data = await fs.promises.readFile(fullPath);
+    const content = data.toString();
+    return content;
+  }
+
+  private createJsons(jsonStrs: string[]): Json[] {
+    const jsons = new Array<Json>();
+    for (const content of jsonStrs) {
+      try {
+        const json = JSON.parse(content);
+        jsons.push(json);
+      } catch (e) {
+        this.logger.Error(e);
+      }
+    }
+    return jsons;
+  }
+
+  /**
+   * Patch to make it works with AJV the existing schema
+   * used for class generation in the demo project.
+   */
+  private cleanUp(schemas: Json) {
+    for (const schema of schemas) {
+      if (schema['$id'] !== undefined && !schema['$id'].endsWith('.json')) {
+        schema['$id'] = `${schema['$id']}.json`;
+      }
+      delete schema['$schema'];
+      if (schema['format'] === 'hostname' || schema['format'] === 'GUID') {
```

Comment:
[FAR] I don't understand why this is needed. `hostname` should be a standard format.
To be investigated.

Reply:
Honestly, it is a pain. I would need to dig in.